### DISCUSSION TITLE
ECONNREFUSED and connect_timeout behaviour

### DISCUSSION THREAD

**Discussion by hardysabs2** [NodeRedis/node_redis#2862] [Q&A]
I had expected connect_timeout  to be a timeout on getting some kind of response to the request for each connection attempt.
However in the case of ECONNREFUSED responding almost immediately (3ms) the timer does not seem to cancel, but remains running.

This means that if you have retry_strategy set up for n retries each with a pause and a maximum overall retry time, depending on the values you may not get through all the retries you have configured because for example at retry 3 of 10 (and not yet reached maximum overall retry time) the uncancelled connect_timeout is reached and you get an error event redis connection in broken state.

All this suggests connect_timeout is not definitely always per connection attempt.
Is there any way around this to make it so?

**Comment by usualdesigner** [comment]
I've seen this too - connectTimeout in node-redis has been a bit inconsistent about being scoped per-attempt vs across the whole reconnect sequence. I wouldn't trust it for a hard overall deadline. What worked for me: track elapsed time yourself inside your reconnectStrategy callback and return an Error once you're past your real deadline - gives you a total-time cutoff you control directly instead of hoping the internal timer cooperates.
